TextToSpeech/impl/TTSSpeaker.cpp

```diff
@@ -673,20 +767,30 @@ void TTSSpeaker::createPipeline() {
         tts_url.append("&text=init");
         std::string LoopbackEndPoint = LOOPBACK_ENDPOINT;
         std::string  LocalhostEndPoint = LOCALHOST_ENDPOINT;
-        //Check if url contains endpoint on localhost, enable PCM audio
-        if((tts_url.rfind(LoopbackEndPoint,0) != std::string::npos)  || (tts_url.rfind(LocalhostEndPoint,0) != std::string::npos)){
-            TTSLOG_INFO("PCM audio playback is enabled");
-            m_pcmAudioEnabled = true;
+
+        if(m_defaultConfig.hasValidLocalEndpoint()) {
```

Comment:
This block seems unnecessary & confusing. Updating the pipeline type can be done from a single place, perhaps in constructURL().

Reply:
This block is needed because createPipeline is called from multiple areas within TTSSpeaker
and need that info for constructing correct pipeline

Reply:
From my understanding, this is the flow of operations
1) You decide which endpoint should be used
2) URL is constructed
3) Check the existing pipeline type & the newly selected endpoints type
4) If they differ reconstruct pipeline, otherwise continue to use the previous one

I think the better place to decide whether PCM Audio should be used / Pipeline type should be in step 1